### PR TITLE
LanceDB Vector Database Integration

### PR DESCRIPTION
**Pull request by TevinWang**
LanceDB Logo

LanceDB is an serverless vector database that makes data management for LLMs frictionless. It has multi-modal search, native Python and JS support, and many ecosystem integrations. Learn more about LanceDB [here](https://lancedb.com).
## Proposed changes
Add LanceDB to the list of vector databases.

We would love to showcase our native python and JS support, so we created folders for each example, with the notebook along with `.py` and `.js` files. Let us know if there are any concerns with this.

### CODE REVIEW

**Comment by TevinWang** [comment]
One thing also, what's the process of updating lancedb availability [here](https://platform.openai.com/docs/guides/embeddings/how-can-i-retrieve-k-nearest-embedding-vectors-quickly) once this PR is merged?

**Comment by AyushExel** [comment]
@ted-at-openai hey can you please take a look at this PR, or let us know if these aren't being accepted for the cookbook. Thanks!

File: examples/vector_databases/lancedb/Youtube_QA_Bot_with_LanceDB_and_OpenAI/main.ipynb [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openai\n",
+    "import os\n",
+    "\n",
+    "# Configuring the environment variable OPENAI_API_KEY\n",
+    "if \"OPENAI_API_KEY\" not in os.environ:\n",
+    "    # OR set the key here as a variable\n",
+    "    openai.api_key = \"sk-...\"\n",
```
Comment (ted-at-openai) [inline examples/vector_databases/lancedb/Youtube_QA_Bot_with_LanceDB_and_OpenAI/main.ipynb:96]: We really try not to encourage hardcoding API keys in code, as it makes it very easy to accidentally leak them. Would ask you to cut this bit before we merge.

Comment (AyushExel) [reply examples/vector_databases/lancedb/Youtube_QA_Bot_with_LanceDB_and_OpenAI/main.ipynb:96]: Thanks ted! @TevinWang can you remove the hardcoding line and just error out if the key is not found in env var?

**Review by ted-at-openai — APPROVED** [review]
If you're happy with it, then happy to merge. Let me know and I'll go ahead.

**Comment by AyushExel** [comment]
Yes @ted-at-openai please do!

**Comment by AyushExel** [comment]
Hey @ted-at-openai 👋 let us know if there's any more changes needed in this. Thanks!

**Review by simonpfish — CHANGES_REQUESTED** [review]
Hi all. Thank you for the contribution. Instead of including the source code for these example apps in this repo, would it be possible to only include the notebooks, and perhaps set up the apps as separate repos and link to them? We'd like to avoid the additional overhead that would come from including those files here.

**Comment by simonpfish** [comment]
Hi all, closing due to inactivity. These apps are awesome. If you guys host them in your own repos I'd love to link to them from the cookbook.

**Pull request closed without merging**